app/src/main/java/meme5/c4q/nyc/meme_project/MainActivity.java

```diff
@@ -1,15 +1,65 @@
 package meme5.c4q.nyc.meme_project;
 
 import android.app.Activity;
+import android.content.Intent;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
 import android.os.Bundle;
-
+import android.view.View;
 
 public class MainActivity extends Activity {
 
+    private static final int REQUEST_IMAGE_CAPTURE = 1;
+    private String imgFilePath;
+    private boolean imageSelected;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        imageSelected = false;
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if(imageSelected){
+            launchChooseMeme();
+        }
+    }
+
+    private void launchChooseMeme(){
+        Intent chooseMeme = new Intent(this,ChooseMemeStyle.class);
+        chooseMeme.putExtra("imgFilePath",imgFilePath);
+        startActivity(chooseMeme);
+    }
+
+    public void takePicture(View view){
+
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
+            imageSelected = true;
```

Comment:
@poojawins if I launch the intent right away, the app crashes due to it trying to launch both the implicit camera intent and the explicit intent for chooseMemeStyle. My work around was to check if a picture was taken after this activity's onResume and then launch the next activity, which is after the phone returns to main activity after the camera intent.

Lines 40 to 42 are necessary to launch the camera intent. You're right about line 43, I'll move it to onActivityResult.